src/routes/users.js

```diff
@@ -1,54 +1,64 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
-const { check, validationResult } = require('express-validator');
-const User = require('../models/User');
+const { check, validationResult } = require("express-validator");
+const User = require("../models/User");
 
 /**
  * @route POST api/users
  * @access Public
  * @desc Register users
- * @todo TODO: Hash password
  */
-router.post('/', [
-  check('name', 'Name is required').not().isEmpty(),
-  check('email', 'Please add a valid email').isEmail(),
-  check('password', 'Please add a password with 6 or more characters').isLength({
-    min: 6
-  }),
-], async (req, res) => {
-
-  // express-validator
-  const errors = validationResult(req);
-  if (!errors.isEmpty()) {
-    return res.status(400).json({errors: errors.array()});
-  }
-  
-  const { name, email, password } = req.body;
-  
-  try {
-    // See if user exists
-    let user = await User.findOne({ email });
-
-    if (user) {
-      return res.status(400).json({errors: [{ msg: `User with ${email} email already exists!` }]});
+
+router.post(
+  "/",
+  [
+    check("name", "Name is required")
+      .not()
+      .isEmpty(),
+    check("email", "Please add a valid email").isEmail(),
+    check(
+      "password",
+      "Please add a password with 6 or more characters"
+    ).isLength({
+      min: 6
+    })
+    //We shouldn't be sending back the attempted password I think
+  ],
+  async (req, res) => {
+    // express-validator
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
     }
 
-    user = new User({
-      name,
-      email,
-      password
-    });
-  
-    // Saves the user to database
-    await user.save();
+    let { name, email, password } = req.body;
```

Comment:
Changed it to a `let` so that way I can hash the password later.  Thinking about it now, I could have left it as `const` and created a new const called `hash` to pass into the password that way we don't mutate the data in any way.

Please comment.